node/narwhal/src/gateway.rs

```diff
@@ -711,9 +711,9 @@ impl<N: Network> Handshake for Gateway<N> {
 
         // Perform the handshake; we pass on a mutable reference to peer_ip in case the process is broken at any point in time.
         let handshake_result = if peer_side == ConnectionSide::Responder {
-            self.handshake_inner_initiator(peer_addr, peer_ip, stream).await
+            self.handshake_inner_initiator(peer_addr, peer_ip, stream).await?
         } else {
-            self.handshake_inner_responder(peer_addr, &mut peer_ip, stream).await
+            self.handshake_inner_responder(peer_addr, &mut peer_ip, stream).await?
         };
 
         // Remove the address from the collection of connecting peers (if the handshake got to the point where it's known).
```

Comment:
looking at it again, shouldn't the connecting peer be removed on handshake failure?

Reply:
Yes, indeed 👍